cylc/flow/task_pool.py

```diff
@@ -2460,129 +2483,28 @@ def log_task_pool(self, log_lvl=logging.DEBUG):
             )
         )
 
-    def filter_task_proxies(
+    def id_match(
```

Comment:
Would it not be easier to remove this entirely and either
* Pass the task pool object as an argument to `id_match.id_match`
* Pass `pool.active_tasks.values()` and `pool.config` to `id_match.id_match`

Reply:
This exists in order to abstract the ID matching code from the task pool.

This is necessary for future plans e.g, https://github.com/cylc/cylc-flow/issues/4357 & https://github.com/cylc/cylc-flow/issues/5677

It also simplifies the `id_match` code which no longer has to juggle itask objects and can focus on pure string manipulation.